security/pkg/nodeagent/vm/nodeagent_v2.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vm
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"istio.io/istio/pkg/log"
+	"istio.io/istio/security/pkg/caclient"
+	caClientInterface "istio.io/istio/security/pkg/nodeagent/caclient/interface"
+	"istio.io/istio/security/pkg/nodeagent/plugin/providers/google/stsclient"
+	pkiutil "istio.io/istio/security/pkg/pki/util"
+	"istio.io/istio/security/pkg/platform"
+	"istio.io/istio/security/pkg/util"
+)
+
+// This file is similar to nodeagent.go, which implements NodeAgent interface.
+// However, this will implement NodeAgent using the new CA protocol which can be found here
+// https://github.com/istio/istio/blob/master/security/proto/istioca.proto
+
+type nodeAgentInternalV2 struct {
+	// Configuration specific to Node Agent
+	config   *Config
+	pc       platform.Client
+	caClient caClientInterface.Client
+	identity string
+	certUtil util.CertUtil
+}
+
+// Start starts the node Agent.
+func (na *nodeAgentInternalV2) Start() error {
```

Comment:
Since you are rewriting the new nodeagent_v2.go, I'm adding several more readability/maintainlibity notes.

- Change nodeagent to citadelAgent. This is consistent with our renaming of the component.  @wattli to confirm.
- Start() needs a lot of refactoring. I'll add advice in a separate thread.

This way, when we retire old implementation, we don't need to redo refactoring/renaming of new implementation again.